Add new api for wikipedia image info

zexigong opened the pull request.

zexigong committed:
Add new api for wikipedia image info

zexigong commented:
https://github.com/Wikidocumentaries/wikidocumentaries-ui/pull/92

tuukka commented:
Thank you for this pull request! I wanted to investigate this more myself but I haven't had the time. Perhaps you can answer some of the questions I have:

1. What if anything does the function `getWikidataImagesFromWikipedia` have to do with Wikidata?
2. Is there a reason to get the information one image at a time, or would it be better to get all images with one request? [Example](https://en.wikipedia.org/w/api.php?action=query&generator=images&titles=Main_Page&prop=imageinfo&iiprop=user|url|extmetadata&redirects=resolve)
3. Is there a reason to return the raw `imageinfo` response, or would it be better to convert the response to the format that the Wikidocumentaries image viewer uses?
4. Would it make sense to share code with `wikimedia-commons.js`, which does similar things?

zexigong commented:
1. It is a typo. Will fix that in the new commit.
2. I decided to get the information one image at a time for two reasons:
- I have tried to get all images with one request. The image metadata returned is not always ordered. There will be extra works to make the metadata ordered.
- Wikipedia get the information one image at a time.
There is no significant latency difference between two approaches. I could try to get the information together if you think it is a better approach.
3. Yes I will convert the response to the format image viewer uses at backend. Previously I did that work at frontend.
4. It is hard to share the entire code here. The urls and required params are different. The converting logic can be shared. I will create a util function in my next commit.
Thanks for your insight!

zexigong committed:
Migrate parsing logic to backend

Migrated parsing logic to backend,
shared code with wikimedia-commons
added more fields
fixed typo

zexigong commented:
Hi @tuukka, in the new commit
- Fixed wikidata typo
- Converted the response to the format that the Wikidocumentaries image viewer uses at the backend
- Shared the code with wikimedia-commons.js by creating a new util function to process the data
- Added more imageviewer fields in the new util function

zexigong committed:
draft

zexigong committed:
Merge remote-tracking branch 'upstream/master'

zexigong committed:
Change the getImageInfoFromWikipedia function to  take a list of image titles and return the corresponding images in the same order; fix the title conversion bug.

zexigong committed:
Revert "draft"

This reverts commit 0c10c2ed4fc68be71054163beaeb97cbecd8520f.

do not need

zexigong committed:
reduce for loop

zexigong committed:
change the nested for loop to consecutive for loop
